drivers/console/uart_mux.c

```diff
@@ -229,6 +229,10 @@ static int uart_mux_init(const struct device *dev)
 {
 	struct uart_mux_dev_data *dev_data = dev->data;
 
+	if (dev == NULL || dev_data == NULL) {
```

Comment:
Here you are checking the null pointer but in the line above, you are dereferencing that pointer without any check. You should do those checks at the beginning of the functions instead.